version: '3'

services:
  wechat:
    build: ./wechat
    volumes:
      - ./data/wechat/config:/app/config
    networks:
      - mynet
    environment:
      TZ: Asia/Shanghai

  frontend:
    build: ./frontend
    ports:
      - "23512:8080"
    networks:
      - mynet
    depends_on:
      - backend
    environment:
      TZ: Asia/Shanghai

  mysql:
    image: mysql:8
    command: --default-authentication-plugin=caching_sha2_password
    environment:
      TZ: Asia/Shanghai
      MYSQL_ROOT_PASSWORD: YOUR_MYSQL_ROOT_PASSWORD
      MYSQL_DATABASE: backend
      MYSQL_USER: django
      MYSQL_PASSWORD: YOUR_MYSQL_PASSWORD
    volumes:
      - ./data/mysql/data:/var/lib/mysql
    networks:
      - mynet
    healthcheck:
      test: [ "CMD", "mysql", "-h", "localhost", "-u", "django", "-pYOUR_MYSQL_PASSWORD", "-e", "SELECT 1" ]
      timeout: 20s
      retries: 10

  backend:
    build: ./backend
    environment:
      TZ: Asia/Shanghai
      OPENAI_API_KEY: YOUR_OPENAI_API_KEY
    depends_on:
      mysql:
        condition: service_healthy
    networks:
      - mynet
    volumes:
      - ./data/backend/config:/opt/tmp/config
networks:
  mynet:
    driver: bridge
